app/controllers/concerns/course/forum/topic_controller_subscription_concern.rb

```diff
@@ -0,0 +1,37 @@
+module Course::Forum::TopicControllerSubscriptionConcern
+  extend ActiveSupport::Concern
+
+  def subscribe
+    if set_subscription_state
```

Comment:
I'd just authorize show in the subscribe action to be safe. You'd also need two sets of rules, but it will be intersection of both.